front/src/common/Map/Layers/OSM.tsx

```diff
@@ -30,36 +33,60 @@ function getMapStyle(mapStyle: string): LayerProps[] {
   }
 }
 
+type FullLayerProps = OrderedLayerProps & { key?: string };
+type ToggledLayers = {
+  showOSM3dBuildings?: boolean;
+};
+
+const filters: Record<string, string> = {
+  batiments_3d: 'showOSM3dBuildings',
+  'building-3d': 'showOSM3dBuildings',
+};
+
 export function genOSMLayerProps(
   mapStyle: string,
+  toggledLayers: ToggledLayers,
   layerOrder?: number
-): (OrderedLayerProps & { key?: string })[] {
+): FullLayerProps[] {
   const osmStyle = getMapStyle(mapStyle);
-  return osmStyle.map((layer) => ({
-    ...layer,
-    key: `${layer.id}-${mapStyle}`,
-    id: `osm/${layer.id}`,
-    layerOrder,
-  }));
+  return osmStyle.reduce<FullLayerProps[]>((acc, layer) => {
+    const isShown = get(toggledLayers, filters[layer.id || ''], true);
```

Comment:
It should be false by default I think
```suggestion
    const isShown = get(toggledLayers, filters[layer.id || ''], false);
```

Reply:
no, it gives `true` for the rest of the OSM layers that are not toggled, so everything except 3d buildings, which are always shown.